### DISCUSSION TITLE
Feature Request: Option to skip `helm diff`

### DISCUSSION THREAD

**Discussion by NfNitLoop** [helmfile/helmfile#931] [Ideas]
I've run into [an issue with a chart][1] that can `helm upgrade` just fine, but fails with an error during `helm diff`.

I'd like an option to skip `helm diff` for that chart entirely, if possible.

I tried `--suppress-diff` but that still runs `helm diff` and fails `helmfile apply`.  As a result, I just can't use this chart with helmfile, and we'll probably have to go back to using bare `helm` commands until it's fixed.

[1]: https://github.com/bitnami/charts/issues/17717

**Comment by yxxhero** [comment]
@NfNitLoop you can use helmfile sync directly.

**Reply by NfNitLoop** [reply]
Hmm, it's not immediately obvious to me what the relationship between `helmfile apply` and `helmfile sync` is from [the docs][1]

```
Available Commands:
  apply        Apply all resources from state file only when there are changes
[...]
  sync         Sync releases defined in state file
```

Is the difference between them really just that `apply` will run `helm diff` and `sync` won't?

[1]: https://helmfile.readthedocs.io/en/latest/#cli-reference

**Reply by NfNitLoop** [reply]
Aha, I see it [here](https://helmfile.readthedocs.io/en/latest/#apply):

> The helmfile apply sub-command begins by executing diff. If diff finds that there is any changes, sync is executed. Adding --interactive instructs Helmfile to request your confirmation before sync.

👍  Thanks!

**Comment by yxxhero** [comment]
see: https://docs.bitnami.com/general/how-to/troubleshoot-helm-chart-issues/#credential-errors-while-upgrading-chart-releases

**Reply by NfNitLoop** [reply]
I've seen that. (It's included in the issue report I wrote/linked.) That error only happens during `helm diff`, not during `helm upgrade`. I assume if it were an actual error it should happen in both cases. But it doesn't happen during `helm upgrade` because *I have* provided the credentials. They're present in a k8s `existingSecret`.

**Reply by yxxhero** [reply]
@NfNitLoop got it.
